components/system-config-mgt/org.wso2.carbon.identity.system.config.mgt/src/main/java/org/wso2/carbon/identity/system/config/mgt/internal/SystemConfigMgtServiceComponent.java

```diff
@@ -0,0 +1,113 @@
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.admin.advisory.mgt.dao.AdminAdvisoryBannerDAO;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.system.config.mgt.advisory.DBBasedAdminBannerDAO;
+
+/**
+ * Service component class for system config mgt Service.
+ */
+@Component(
+        name = "identity.system.config.mgt.component",
+        immediate = true
+)
+public class SystemConfigMgtServiceComponent {
+
+    private static final Log log = LogFactory.getLog(SystemConfigMgtServiceComponent.class);
+
+    /**
+     * Activate the SystemConfigMgtServiceComponent.
+     *
+     * @param context
+     */
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            BundleContext bundleContext = context.getBundleContext();
+            // Register the Database based Admin Advisory Banner DAO.
+            bundleContext.registerService(AdminAdvisoryBannerDAO.class, new DBBasedAdminBannerDAO(), null);
+
+            if (log.isDebugEnabled()) {
+                log.debug("System Config Mgt Service Component is activated.");
+            }
+        } catch (Throwable e) {
+            log.error("Error while activating System Config management service.", e);
+        }
+    }
+
+    /**
+     * Deactivate the CORSManagementServiceComponent.
+     *
+     * @param context
+     */
+    @Deactivate
+    protected void deactivate(ComponentContext context) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("CORSManagementServiceComponent is deactivated.");
+        }
+    }
+
+    /**
+     * Set the ConfigurationManager.
+     *
+     * @param configurationManager The {@code ConfigurationManager} instance.
+     */
+    @Reference(
+            name = "resource.configuration.manager",
+            service = ConfigurationManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unregisterConfigurationManager"
+    )
+    protected void registerConfigurationManager(ConfigurationManager configurationManager) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Registering the ConfigurationManager in CORSManagementService.");
+        }
+        SystemConfigMgtServiceHolder.getInstance().setConfigurationManager(configurationManager);
+    }
+
+    /**
+     * Unset the ConfigurationManager.
+     *
+     * @param configurationManager The {@code ConfigurationManager} instance.
+     */
+    protected void unregisterConfigurationManager(ConfigurationManager configurationManager) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Unregistering the ConfigurationManager in CORSManagementService.");
```

Comment:
remove `CORSManagementService`

Reply:
also we dont need to have if check here, since the debug log contains a static string.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/compare/285ff9cce4eea45be0097bd98637327b117b6542..d66d7869f528d604f1cba38076c31122a7aea260